zaza/openstack/charm_tests/neutron/tests.py

```diff
@@ -29,6 +30,206 @@
 import zaza.openstack.utilities.openstack as openstack_utils
 
 
+class NeutronGatewayTest(test_utils.OpenStackBaseTest):
+    """Test basic Neutron Gateway Charm functionality."""
+
+    @classmethod
+    def setUpClass(cls):
+        """Run class setup for running Neutron Gateway tests."""
+        super(NeutronGatewayTest, cls).setUpClass()
+        cls.current_os_release = openstack_utils.get_os_release()
+        cls.services = cls._get_services()
+
+        # set up clients
+        cls.neutron_client = (
+            openstack_utils.get_neutron_session_client(cls.keystone_session))
+
+        bionic_stein = openstack_utils.get_os_release('bionic_stein')
+
+        cls.pgrep_full = (True if cls.current_os_release >= bionic_stein
+                          else False)
+
+    def test_400_create_network(self):
```

Comment:
Not a blocker, but the 4xx tests feel much more about testing the neutron-api than the gateway. Ironically NeutronNetworkingTest is much more about the gateway. It might be good to add these to the NeutronAPITest class rather than this one.

Reply:
done - [commit](https://github.com/openstack-charmers/zaza-openstack-tests/pull/142/commits/55fc718f981f22cb7a555de993960841d8d5196a)

Validated via func-smoke on both neutron-api and neutron-gateway